cartridge/shop/managers.py

```diff
@@ -22,26 +22,20 @@ def from_request(self, request):
         cart.
         """
         cart_id = request.session.get("cart", None)
-        cart = None
-        if cart_id:
+        cart = self.current().filter(id=cart_id)
+        last_updated = now()
+        # Update timestamp and clear out old carts.
+        if cart_id and cart.update(last_updated=last_updated):
+            self.expired().delete()
+        else:
```

Comment:
Wasn't it modified even if there was no cart? if `cart_id == None` then `cart = None` would remain untouched, triggering the `if not cart` branch, which tries to delete any order step that might exist.